packages/fleather/lib/src/widgets/controller.dart

```diff
@@ -171,7 +171,7 @@ class FleatherController extends ChangeNotifier {
   // document; `false` otherwise
   bool _captureAutoFormatCancellationOrUndo(
       ParchmentDocument document, int position, int length, Object data) {
-    if (!_autoFormats.hasActiveSuggestion) return true;
+    if (!_autoFormats.canUndo) return true;
```

Comment:
If active suggestions with empty undo, we still need to cancel active suggestion I think